### DISCUSSION TITLE
Parse/process options to control addition of ';' and '\n'

### DISCUSSION THREAD

**Discussion by ShayRubach** [swc-project/swc#3388] [Q&A] [answered]
Hey. I have an input
```javascript
var x = y
```
that is simply going through `parse_js` -> `process_js` and its output is
```javascript
var x = y;\n
```

Which options control this formatting? I went over `swc::config::Options` and its inner members and didn't find any hints about `;` or `\n` in particular.
I basically want `swc` to make zero optimizations/changes/additions that are not invoked by my visitor during transformation. 
Thanks.

**Comment by kdy1** [comment, marked as answer]
There's no option to do so.

**Reply by ShayRubach** [reply]
I see.
Could you direct me to the code who appends them? Maybe I can PR it if in my free time 😄 
Thanks for the quick response.

**Reply by RiESAEX** [reply]
I think is emit by https://github.com/swc-project/swc/blob/3eaebdf479ff24ceccac2bcfb2e6e65ee0e5d413/crates/swc_ecma_codegen/src/macros.rs#L83
